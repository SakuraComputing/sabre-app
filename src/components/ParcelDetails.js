import React from 'react';
import Address from "./Address";

const ParcelDetails = ({ parcel, address }) => (
    <table className='parcel-details' frameBorder="1">
        <tbody>
            <tr>Address:</tr>
            <tr className="address-details" >{address ? <Address address={address}/> : 'Address not found'}</tr>
            <tr>Service Code: {parcel.serviceCode}</tr>
        </tbody>
    </table>
);
export default ParcelDetails;
