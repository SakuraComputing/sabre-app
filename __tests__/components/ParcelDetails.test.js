import React from 'react';
import { shallow } from 'enzyme';
import ParcelDetails from '../../src/components/ParcelDetails';

const parcel = {
    parcelId: "JD0002256504047335",
    addressNo: 1,
    serviceCode: "HOME24",
    latestScan: 1509494399,
    createdAt: 1506816000
};

const address = {
    addressNo: 1,
    addressName: "Alex Delarge",
    addressLine1: "Broadmoor",
    addressLine2: "",
    town: "Crowthorne",
    county: "Berkshire",
    telephone: "01344 77 3111"
};

describe('parcel details',() => {
    it('should render parcel and address',() => {
        const wrapper = shallow(<ParcelDetails parcel={parcel} address={address}/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('should render parcel and display no address found',() => {
        const wrapper = shallow(<ParcelDetails parcel={parcel} />);
        const expenseSummaryField = wrapper.find('.address-details').text();
        expect(expenseSummaryField).toEqual('Address not found');
    });
});


