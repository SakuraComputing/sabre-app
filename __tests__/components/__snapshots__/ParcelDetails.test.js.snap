// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parcel details should render parcel and address 1`] = `
<table
  className="parcel-details"
  frameBorder="1"
>
  <tbody>
    <tr>
      Address:
    </tr>
    <tr
      className="address-details"
    >
      <Address
        address={
          Object {
            "addressLine1": "Broadmoor",
            "addressLine2": "",
            "addressName": "Alex Delarge",
            "addressNo": 1,
            "county": "Berkshire",
            "telephone": "01344 77 3111",
            "town": "Crowthorne",
          }
        }
      />
    </tr>
    <tr>
      Service Code: 
      HOME24
    </tr>
  </tbody>
</table>
`;
